service/controller/resource/monitoring/verticalpodautoscaler/resource.go

```diff
@@ -214,6 +209,12 @@ func (r *Resource) getMaxMemory(nodes *v1.NodeList) (*resource.Quantity, error)
 		return nil, microerror.Mask(err)
 	}
 
+	// Memory must be a whole number of bytes
+	q.Set(int64(q.AsApproximateFloat64()))
```

Comment:
Can we not use https://github.com/kubernetes/apimachinery/blob/f357b1fa74b71ca0cffc090e447983950c2bafc5/pkg/api/resource/quantity.go#L483 ?

Reply:
I don't know how, and I guess it would be more complex.

It says "AsInt64 returns a representation of the current value as an int64 if a fast conversion is possible. If false is returned, callers must use the inf.Dec form of this quantity."
Long story short: as far as I understand these functions, they don't want to lose data/precision.
So I have to do it myself, with this conversion.

Reply:
Sure then let's go